frontend/src/views/Game.tsx

```diff
@@ -1,10 +1,28 @@
 import Editor from '@monaco-editor/react';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Room } from '../api/Room';
+
+type LocationState = {
+  room: Room,
+}
 
 function GamePage() {
+  const location = useLocation<LocationState>();
+  const [room, setRoom] = useState<Room | null>(null);
+
+  // Called every time location changes
+  useEffect(() => {
+    if (location && location.state && location.state.room) {
```

Comment:
What causes the `location` to lose its value? I saw that if you refresh the page (incognito, cache-clearing, or regular), it retains the value of `location.state.room`. If you instead load it on a different tab or change URLs then return, the `location.state` value is `null`.

This is fantastic, and likely exactly the behavior we want; however, since I don't understand this variable that well, I wanted to ask -- are there are any edge cases with this we should watch out for?

Reply:
Building on this, I was going to ask whether we should store URLs on the game page as `/game?roomId=XXXXXX`. It doesn't seem like that's needed here, but it may help with sharing the URL / code easily, as otherwise the other users have no other option but to type it out. Thoughts?

Reply:
I'm not sure having unique URLs for rooms would provide tangible benefit. To share it you could send just the code, and others can copy/paste that into the "join room" box without typing it.

Reply:
That's true, I still think it would be easy though to just share the URL -- that's copying only one thing vs copying two. Plus, in this setup I think it would help to provide both options (there isn't any choice overload that happens here, so whichever method people prefer can be used). I guess overall, it's up to @alankbi.

Reply:
I think that's something that might make sense to implement on the join page; the way I imagined it, the host on the waiting room page can copy a share link such as "/game/join?id=104020" and share that with others, and then when people navigate to that link, it'd basically be the join page but skipping past the enter room id step straight to the enter nickname step. But with all that, I added a [card](https://trello.com/c/SlNjdzCe/101-join-page-takes-query-param-with-roomid-for-quick-sharing) to the backlog so that we can discuss this further.

Reply:
To answer the first question, this is a quote from the documentation: 

>The useLocation hook returns the location object that represents the current URL. You can think about it like a useState that returns a new location whenever the URL changes.

So I guess refreshing doesn't change the URL but if you go forward/back and all that then it does?

Reply:
Thanks for the answer to the `useLocation` hook. As for this:
> it'd basically be the join page but skipping past the enter room id step straight to the enter nickname step.

That's how I imagined it too, so it let's users join the game more quickly.